ambari-server/src/main/java/org/apache/ambari/server/state/ConfigImpl.java

```diff
@@ -378,8 +378,19 @@ void persistEntitiesInTransaction(ClusterConfigEntity entity) {
    * {@inheritDoc}
    */
   @Override
-  @Transactional
   public void save() {
+    ClusterConfigEntity entity = saveWithoutPublishing();
```

Comment:
Is this a general contract to save and then publish (comment)? Is the saveWithoutPublishing also persisting the entity?
We should make sure the expected behavior is standardized so in future, anybody doing a save and publish follows the exact same semantics.